core/app/models/spree/taxon.rb

```diff
@@ -105,6 +106,14 @@ def child_index=(idx)
 
     private
 
+    def sync_taxonomy_name
+      if saved_change_to_name? && root?
+        return if taxonomy.name.to_s == name.to_s
+
+        taxonomy.update(name: name)
```

Comment:
@MatthewKennedy if we update the taxonomy this will also run 

https://github.com/spree/spree/blob/a0137d7aa25fa48639c88e5baef11bd236559812/core/app/models/spree/taxonomy.rb#L27

so we would need to also add a check if the names are equal in that method

Reply:
@MatthewKennedy what happens with all of the taxon's slugs when we update the parent name which will update the slug(?)

Reply:
Slugs are based off the permalinks, not the names so they shouldn't be affected.